includes/Modules/Analytics.php

```diff
@@ -1385,9 +1419,12 @@ protected function create_analytics_site_data_request( array $args = array() ) {
 			$dimension_filter->setOperator( 'EXACT' );
 			$args['page'] = str_replace( trim( $this->context->get_reference_site_url(), '/' ), '', esc_url_raw( $args['page'] ) );
 			$dimension_filter->setExpressions( array( rawurldecode( $args['page'] ) ) );
+		}
+
+		if ( ! empty( $dimension_filters_clauses ) ) {
 			$dimension_filter_clause = new Google_Service_AnalyticsReporting_DimensionFilterClause();
 			$dimension_filter_clause->setFilters( array( $dimension_filter ) );
-			$request->setDimensionFilterClauses( array( $dimension_filter_clause ) );
+			$request->setDimensionFilterClauses( $dimension_filters_clauses );
```

Comment:
This is still not quite right (no better place to leave this comment although this specific line _is_ correct).

The `$dimension_filter_clause` immediately above belongs in the preceding conditional where the filter is defined.

This conditional should only have the call to `$request->setDimensionFilterClauses()`.